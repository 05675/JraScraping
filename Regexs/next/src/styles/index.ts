export { StyledButton } from './button';
export { StyledInput } from './input';
export { StyledLabel } from './label';
export * from './listCommon';
