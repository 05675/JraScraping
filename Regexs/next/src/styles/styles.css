body {
  margin: 0;
  font-family: Noto Sans JP, Hiragino Sans, Meiryo, sans-serif;
  background-color: #edf4fa;
}

button,
input,
select,
textarea {
  font-family: inherit;
}

.common-header {
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
  height: 48px;
  position: fixed;
}

.page-header-task {
  top: 48px;
  left: 0;
  right: 0;
  z-index: 1;
  background: #edf4fa;
  height: 72px;
  position: fixed;
}

.page-header-life-insurance {
  top: 48px;
  left: 0;
  right: 0;
  z-index: 1;
  position: fixed;
}
