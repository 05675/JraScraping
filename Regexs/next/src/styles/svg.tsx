import styled, { css } from 'styled-components';
import * as svgs from '@assets/images';

interface StyledSVGProps {
  readonly color?: string;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const doStyleSVG = (svg: any) => styled(svg)<StyledSVGProps>`
  ${props =>
    css`
      path {
        fill: ${props.color || '#949494'};
      }
    `}
`;

const {
  ArrowDownSVG,
  ArrowLeftSVG,
  CheckBadgeSVG,
  CheckSVG,
  CloseSVG,
  ErrorSVG,
  EyeOffSVG,
  EyeSVG,
  LockSVG,
  LogoWhiteSVG,
  LogoSVG,
  MenuSVG,
  NotificationsSVG,
  VectorSVG,
  PlusSVG,
} = svgs;

export const StyledArrowDownSVG = doStyleSVG(ArrowDownSVG);
export const StyledArrowLeftSVG = doStyleSVG(ArrowLeftSVG);
export const StyledCheckBadgeSVG = doStyleSVG(CheckBadgeSVG);
export const StyledCheckSVG = doStyleSVG(CheckSVG);
export const StyledCloseSVG = doStyleSVG(CloseSVG);
export const StyledErrorSVG = doStyleSVG(ErrorSVG);
export const StyledEyeOffSVG = doStyleSVG(EyeOffSVG);
export const StyledEyeSVG = doStyleSVG(EyeSVG);
export const StyledLockSVG = doStyleSVG(LockSVG);
export const StyledLogoWhiteSVG = doStyleSVG(LogoWhiteSVG);
export const StyledLogoSVG = doStyleSVG(LogoSVG);
export const StyledMenuSVG = doStyleSVG(MenuSVG);
export const StyledNotificationsSVG = doStyleSVG(NotificationsSVG);
export const StyledVectorSVG = doStyleSVG(VectorSVG);
export const StyledPlusSVG = doStyleSVG(PlusSVG);
