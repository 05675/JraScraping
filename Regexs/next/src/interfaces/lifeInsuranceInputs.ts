import { InsuranceCategories } from '@src/model/entity';
import { ErrorInfo } from '@src/interfaces/errorInfo';
import { InputProps } from '@src/styles';

export interface NenchoInsuranceLifeInput2020 {
  id?: number;
  nenchoId: string;
  category: InsuranceCategories;
  firmName: string;
  categoryDetail: string;
  period: string;
  contractorName: string;
  receiverName: string;
  relation: string;
  payment: string;
}

type ValidateFormProps = {
  isError: boolean;
  errorMessage: string;
};

export interface NenchoInsuranceLifeInput2020Validate {
  category: ValidateFormProps;
  firmName: ValidateFormProps;
  categoryDetail: ValidateFormProps;
  period: ValidateFormProps;
  contractorName: ValidateFormProps;
  receiverName: ValidateFormProps;
  relation: ValidateFormProps;
  payment: ValidateFormProps;
}

export interface OptionType {
  key: string;
  label: string;
  value: string | boolean;
}

export interface InputWithLabelProps extends InputProps {
  label: string;
  name: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  errorMessage?: string;
}

export interface SelectBoxProps {
  label: string;
  options: OptionType[];
  value: OptionType['value'];
  onChange: (selectedOption: OptionType) => void;
}

export interface LifeInsuranceInputsProps {
  isUpdatePage: boolean;
  nenchoId: string | string[];
  lifeInsuranceData?: NenchoInsuranceLifeInput2020;
  token?: string;
  errorInfo?: ErrorInfo;
}
