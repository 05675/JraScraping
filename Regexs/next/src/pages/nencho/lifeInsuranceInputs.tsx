import React, { useState } from 'react';
import { NextPage } from 'next';
import { StyledLabel, StyledText } from '@src/styles/label';
import { SingleSelectComponent } from '@src/components/singleSelection';
import { StyledInput } from '@src/styles';
import { withAuthSync } from '@src/util/auth';
import { PageHeader } from '@src/components/pageHeader';

type OptionType = {
  label: string;
  value: string;
  groupId?: string;
};

// TODO:リストアイテム仮置き
interface ListItem {
  item: OptionType[];
}

const Form: React.FC<{ label: string }> = ({ label }) => {
  return (
    <>
      <StyledLabel>{label}</StyledLabel>
      <StyledInput style={{ margin: '6px auto 16px auto' }} />
    </>
  );
};

const ListBox: React.FC<{ label: string; items: ListItem }> = ({ label, items }) => {
  return (
    <>
      <StyledLabel> {label} </StyledLabel>
      <SingleSelectComponent options={items.item} onChange={useState} />
    </>
  );
};

/**
 * 生命保険入力画面
 *
 * @returns {*} Reactコンポーネント
 */
const LifeInsuranceInputs: NextPage = () => {
  const saveLifeInsuranceInfo = () => {
    // FIXME: 2020-02-26 K.TANAKA DBへの登録処理をここに実装する
    // eslint-disable-next-line no-alert
    alert('入力内容をDBに登録');
  };

  const insList: ListItem = {
    item: [],
  };
  // const [insuranceType, setInsuranceTypeList] = useState();
  return (
    <>
      <PageHeader title='生命保険' buttonName='保存' buttonFunc={() => saveLifeInsuranceInfo()} />

      <div style={{ margin: '0 16px' }}>
        <ListBox label='生命保険の種類' items={insList} />
        <Form label='保険会社当の名称' />
        <Form label='保険等の種類' />
        <Form label='保険期間' />
        <Form label='保険等の契約者の氏名' />
        <StyledLabel>保険金等の受取人</StyledLabel>
        <div style={{ display: 'flex' }}>
          <div style={{ width: '50%' }}>
            <Form label='氏名' />
          </div>
          <div style={{ width: '50%', marginLeft: '16px' }}>
            <Form label='あなたとの続柄' />
          </div>
        </div>
        <ListBox label='新・旧の区分' items={insList} />
        <Form label='申告額' />
        <StyledText>
          あなたが本年中に支払った保険料等の金額（分配を受けた剰余金等の控除後の金額）
        </StyledText>
      </div>
    </>
  );
};

export default withAuthSync(LifeInsuranceInputs);
