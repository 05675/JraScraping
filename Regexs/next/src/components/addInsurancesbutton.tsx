import React from 'react';
import { StyledButton } from '@src/styles';
import { StyledPlusSVG } from '@src/styles/svg';

type AddInsurancesButtonProps = {
  label?: string;
  important?: boolean;
  primary?: boolean;
  warning?: boolean;
  smallSize?: boolean;
  fontSize?: string;
  width?: string;
  height?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onClick?: (event: any) => void;
};

export const AddInsurancesButton: React.FC<AddInsurancesButtonProps> = props => {
  return (
    <>
      <div
        style={{
          position: 'relative',
          width: `${props.width ?? '100%'}`,
          height: `${props.height ?? '48px'}`,
        }}>
        <span className='plus-icon'>
          <StyledPlusSVG color='#525252' />
        </span>
        <StyledButton
          important={props.important}
          primary={props.primary}
          warning={props.warning}
          smallSize={props.smallSize}
          fontSize={props.fontSize}
          width={props.width}
          height={props.height}
          onClick={props.onClick}
          labelOffsetLeft='48px'
          noShadow>
          {props.children}
        </StyledButton>
      </div>

      <style jsx>
        {`
          .plus-icon {
            position: absolute;
            width: 24px;
            height: 24px;
            top: calc(50% - 12px);
            left: 24px;
          }
        `}
      </style>
    </>
  );
};

export default AddInsurancesButton;
