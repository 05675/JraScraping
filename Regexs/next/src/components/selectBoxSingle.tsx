import React from 'react';
import Select, { components, Styles } from 'react-select';
import { StyledArrowDownSVG } from '@src/styles/svg';

/**
 * コンボボックスのデザイン指定
 */
const customStyles: Styles = {
  control: (styles, state) => ({
    ...styles,
    width: '100%',
    height: '48px',
    color: '#525252',
    background: 'rgba(1,1,1,0)',
    borderRadius: '0px',
    fontSize: '16px',
    fontStyle: 'normal',
    lineHeight: '180%',
    letterSpacing: '0.06em',
    border: 'none',
    borderBottom: '1px solid',
    borderBottomColor: (() => {
      if (state.selectProps.isError) {
        return '#fa3939';
      }
      if (state.isFocused) {
        return '#3aa0fa';
      }
      return '#ebebeb';
    })(),
    boxShadow: (() => {
      if (state.selectProps.isError && state.isFocused) {
        return '0px 1px 0px #fa3939';
      }
      if (state.isFocused) {
        return '0px 1px 0px #3aa0fa';
      }
      return 'none';
    })(),
  }),

  option: (styles, state) => ({
    ...styles,
    color: '#525252',
    backgroundColor: state.isSelected ? '#ffffff' : '#ffffff',
  }),

  indicatorSeparator: () => ({}),

  valueContainer: styles => ({
    ...styles,
    padding: '0',
  }),

  singleValue: styles => ({
    ...styles,
    margin: '0',
  }),
};

/**
 * ドロップダウンのコンポーネント指定
 *
 * @param {any} props ドロップダウンインジケータの情報
 * @returns {*} ドロップダウンコンポーネント
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const DropdownIndicator: React.FC<any> = props => {
  return (
    <>
      <div style={{ marginRight: props.selectProps.isError ? '40px' : '0' }}>
        <components.DropdownIndicator {...props}>
          <StyledArrowDownSVG color='#949494' width='12px' height='12px' />
        </components.DropdownIndicator>
      </div>
    </>
  );
};

type OptionType = {
  key: string;
  label: string;
  value: string | boolean;
};
/**
 * react-selectの単一選択ドロップダウンメニュー用インターフェース
 *
 * @interface SelectBoxSingleProps
 */
export interface SelectBoxSingleProps {
  id?: string;
  placeholder?: string;
  options: OptionType[];
  value: OptionType['value'];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onChange: (value: any) => void;
  isError?: boolean;
}

/**
 * react-selectを利用した単一選択ドロップダウンメニューの共通コンポーネント
 *
 * @param {SelectBoxSingleProps} props {
 *     placeholder
 *     value
 *     options
 *     onChange
 * }
 * @returns {*} Reactコンポーネント
 */
export const SelectBoxSingle: React.FC<SelectBoxSingleProps> = ({
  placeholder,
  options,
  value,
  ...props
}) => {
  return (
    <Select
      isSearchable={false}
      styles={customStyles}
      components={{ DropdownIndicator }}
      placeholder={placeholder || '選択してください'}
      value={options?.filter(option => option.value === value)}
      options={options}
      isError={props.isError}
      {...props}
    />
  );
};
