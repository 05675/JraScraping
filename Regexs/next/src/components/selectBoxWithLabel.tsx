import React from 'react';
import { StyledLabel, StyledErrorText } from '@src/styles';
import { StyledErrorSVG } from '@src/styles/svg';
import { SelectBoxSingle, SelectBoxSingleProps } from '@src/components/selectBoxSingle';

export interface SelectBoxWithLabelBoxProps extends SelectBoxSingleProps {
  label: string;
  name: string;
  errorMessage?: string;
}

export const SelectBoxWithLabel: React.FC<SelectBoxWithLabelBoxProps> = props => {
  return (
    <div style={{ marginBottom: '32px' }}>
      <div className='label'>
        <StyledLabel id='titleLabel'>{props.label}</StyledLabel>
      </div>
      <div style={{ position: 'relative' }}>
        <SelectBoxSingle id='selectBoxSingle' {...props} />
        {props.isError && (
          <span className='errorIcon'>
            <StyledErrorSVG id='errorIcon' color='#fa3939' />
          </span>
        )}
      </div>
      {props.isError && (
        <div className='errorText'>
          <StyledErrorText id='errorText'>{props.errorMessage}</StyledErrorText>
        </div>
      )}

      {/* style */}
      <style jsx>{`
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
          background: white;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      `}</style>
    </div>
  );
};
