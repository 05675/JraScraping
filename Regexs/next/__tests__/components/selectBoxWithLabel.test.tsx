/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { SelectBoxWithLabel, SelectBoxWithLabelBoxProps } from '@src/components/selectBoxWithLabel';
import { StyledLabel, StyledErrorText } from '@src/styles';
import { StyledErrorSVG } from '@src/styles/svg';
import { SelectBoxSingle } from '@src/components/selectBoxSingle';
import { OptionType } from '@src/interfaces/lifeInsuranceInputs';

const handleChange: (value: any) => void = jest.fn();

const categoryOptions: OptionType[] = [
  { key: 'category', label: '一般の生命保険：新制度', value: 'insurance_0001' },
  { key: 'category', label: '一般の生命保険：旧制度', value: 'insurance_0002' },
  { key: 'category', label: '介護医療保険：新制度', value: 'insurance_0003' },
  { key: 'category', label: '個人年金保険：新制度', value: 'insurance_0004' },
  { key: 'category', label: '個人年金保険：旧制度', value: 'insurance_0005' },
];

const propsDefault: SelectBoxWithLabelBoxProps = {
  label: 'labelData',
  name: 'nameData',
  placeholder: 'placeholderData',
  options: categoryOptions,
  value: 'valueData',
  onChange: handleChange,
  errorMessage: 'errerMessageData',
  isError: true,
};

const propsBlank: SelectBoxWithLabelBoxProps = {
  label: '',
  name: '',
  placeholder: '',
  options: categoryOptions,
  value: '',
  onChange: handleChange,
  errorMessage: '',
  isError: false,
};

describe('snapshot', () => {
  test('default', () => {
    const snapshot = renderer.create(<SelectBoxWithLabel {...propsDefault} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('StyledLabel タイトルラベル', () => {
  test('初期表示', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsDefault} />);
    const target = nextPage.find(StyledLabel).find('#titleLabel');
    expect(target.length).toBe(1);

    expect(target.text()).toMatch('labelData');
  });
});

describe('SelectBoxSingle セレクトボックス', () => {
  test('初期表示', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsDefault} />);
    const target = nextPage.find(SelectBoxSingle).find('#selectBoxSingle');
    expect(target.length).toBe(1);

    expect(target.prop('label')).toMatch('labelData');
    expect(target.prop('name')).toMatch('nameData');
    expect(target.prop('placeholder')).toMatch('placeholderData');
    expect(target.prop('options')).toEqual(categoryOptions);
    expect(target.prop('value')).toMatch('valueData');
    expect(target.prop('onChange')).toEqual(handleChange);
    expect(target.prop('errorMessage')).toMatch('errerMessageData');
    expect(target.prop('isError')).toBeTruthy();
  });
});

describe('StyledErrorSVG エラーアイコン', () => {
  test('初期表示：isError=true', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsDefault} />);
    const target = nextPage.find(StyledErrorSVG).find('#errorIcon');
    expect(target.length).toBe(1);
  });
  test('初期表示：isError=false', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsBlank} />);
    const target = nextPage.find(StyledErrorSVG).find('#errorIcon');
    expect(target.length).toBe(0);
  });
});

describe('StyledErrorText エラーテキスト', () => {
  test('初期表示：isError=true', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsDefault} />);
    const target = nextPage.find(StyledErrorText).find('#errorText');
    expect(target.length).toBe(1);

    expect(target.text()).toMatch('errerMessageData');
  });
  test('初期表示：isError=false', () => {
    const nextPage = shallow(<SelectBoxWithLabel {...propsBlank} />);
    const target = nextPage.find(StyledErrorText).find('#errorText');
    expect(target.length).toBe(0);
  });
});
