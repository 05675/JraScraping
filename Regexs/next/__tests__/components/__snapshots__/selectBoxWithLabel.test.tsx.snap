// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot default 1`] = `
<div
  style={
    Object {
      "marginBottom": "32px",
    }
  }
>
  <div
    className="label"
  >
    <label
      className="sc-AxhCb fZxPgc"
      id="titleLabel"
    >
      labelData
    </label>
  </div>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className=" css-2b097c-container"
      id="selectBoxSingle"
      onKeyDown={[Function]}
    >
      <div
        className=" css-y6gifx-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-86rg1c-ValueContainer"
        >
          <div
            className=" css-1wa3eu0-placeholder"
          >
            placeholderData
          </div>
          <input
            aria-autocomplete="list"
            className="css-62g3xt-dummyInput"
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            tabIndex="0"
            value=""
          />
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-62wm9a-IndicatorSeparator"
          />
          <div
            style={
              Object {
                "marginRight": "40px",
              }
            }
          >
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg>

                className="sc-fzokOt enYZSd"
                color="#949494"
                height="12px"
                width="12px"
              />
            </div>
          </div>
        </div>
      </div>
      <input
        name="nameData"
        type="hidden"
        value=""
      />
    </div>
    <span
      className="errorIcon"
    >
      <svg>

        className="sc-fzoXzr gTpTcX"
        color="#fa3939"
        id="errorIcon"
      />
    </span>
  </div>
  <div
    className="errorText"
  >
    <div
      className="sc-AxheI cnbkls"
      id="errorText"
    >
      errerMessageData
    </div>
  </div>
  <style
    jsx={true}
  >
    
        .label {
          height: 18px;
          margin: 0;
        }
        .errorIcon {
          position: absolute;
          width: 24px;
          height: 24px;
          top: 12px;
          right: 0px;
          background: white;
        }
        .errorText {
          width: 100%;
          left: 0px;
          margin-top: 6px;
        }
      
  </style>
</div>
`;
